### PR TITLE
Feature/event filters

### PR DESCRIPTION
**Pull request by laij84**
Can you both run the `yarn lint` and check this works for both of you - making it sure it works on both your ide and normal terminal

### CODE REVIEW

File: src/components/AppContext.js [hunk 1]
```diff
@@ -51,20 +61,70 @@ class Provider extends Component {
   }
 
   clearFilters = () => {
-    this.setState(initialState)
+    const { state } = this
+    state.filterOpen = null
+    state.filters = {
```
Comment (ds300) [inline src/components/AppContext.js]: Mutation of `state` like this is not so kosher in react apps. We could make it safe by using `immer` but I think in this case sticking to the more idiomatic functional style would be fine.

i.e.

```js
this.setState({
  ...this.state,
  filterOpen: null,
  filters: initialState.filters,
})
```

File: src/components/AppContext.js [hunk 2]
```diff
@@ -51,20 +61,70 @@ class Provider extends Component {
...
+      area: [],
+      timeOfDay: [],
+    }
+    this.setState(state)
+  }
+
+  closeSiblingFilters = (filterName, isOpen) => {
+    if (isOpen && filterName != this.state.openFilter) {
+      const { state } = this
+      this.state.filterOpen = filterName
```
Comment (ds300) [inline src/components/AppContext.js]: There's more state mutation here. You can do this instead:
```js
this.setState({...this.state, filterOpen: filterName})
```

File: src/components/CheckboxSet.js [hunk 3]
```diff
@@ -44,27 +45,29 @@ class CheckboxSet extends Component {
...
-                      e,
-                      this.props.filterName
-                    )
-                  }
-                />
-              </ListItem>
-            ))}
+            {constants[this.props.filterName].map(option => {
+              return (
+                <ListItem key={shortid.generate()}>
```
Comment (ds300) [inline src/components/CheckboxSet.js]: Nit: these will be remounted on every render. It won't cause perf issues in this case, but if we weren't pressed for time I'd recommend using `this.makeId(option)` as the key here too.

File: src/components/CheckboxSet.js [hunk 4]
```diff
@@ -44,27 +45,29 @@ class CheckboxSet extends Component {
...
+                  <Checkbox
+                    checked={
+                      context.state.filters[this.props.filterName].indexOf(
+                        option
+                      ) >= 0
+                    }
+                    label={option}
+                    value={option}
+                    id={`${this.makeId(option)}`}
+                    name={`${this.makeId(option)}`}
```
Comment (ds300) [inline src/components/CheckboxSet.js]: nit: these don't need to be in template strings

File: src/components/navigation/presentational/navItems.js [hunk 5]
```diff
@@ -34,8 +35,8 @@ const NavItems = props => {
       <Column flex={3} alignItems="center" justifyContent="center">
         <img src={logo} alt="" />
       </Column>
-      {listItems.map((items, index) => (
-        <Column key={index}>
+      {listItems.map(items => (
+        <Column key={shortid.generate()}>
```
Comment (ds300) [inline src/components/navigation/presentational/navItems.js]: nit: these again will be remounted on every render. Not sure why the index wasn't just fine? Nav items are static anyway, right?

File: src/layouts/index.js [hunk 6]
```diff
@@ -34,14 +34,14 @@ const Layout = props => (
             cta: 'donate',
           }}
         />
-        <div>{props.children()}</div>
-      </div>
+        <main>{props.children()}</main>
```
Comment (ds300) [inline src/layouts/index.js]: Wow! I had no idea there was a `main` tag in html! And I thought I'd seen it all! 😅

**Comment by nats12** [comment]
All good with the lint in terminal. Checking the 'show only free events' box expands other drop downs though

File: src/components/AppContext.js (hunk 1 above)
Comment (laij84) [reply src/components/AppContext.js]: Spent a few hours on saturday figuring out this problem, but when you update state it also mutates the initial state object, so you cannot use it to reset state

File: src/components/navigation/presentational/navItems.js (hunk 5 above)
Comment (laij84) [reply src/components/navigation/presentational/navItems.js]: as discussed we'll leave this for now as using index is antipattern

**Pull request merged**